src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java

```diff
@@ -283,4 +301,109 @@ public void uploadFileUnsupportedFileExtension() throws Exception {
         // upload file
         request.postWithMultipartFile("/api/fileUpload?keepFileName=true", file.getOriginalFilename(), "file", file, JsonNode.class, HttpStatus.BAD_REQUEST);
     }
+
+    @Test
+    @WithMockUser(value = "instuctor1", roles = "INSTRUCTOR")
+    public void testGetLecturePdfAttachmentsMerged_InvalidLectureId() throws Exception {
+        // get access token and then send request using the access token
+        String accessToken = request.get("/api/files/attachments/access-token/merge-pdf", HttpStatus.OK, String.class);
+        request.get("/api/files/attachments/lecture/" + 999999999 + "/merge-pdf?access_token=" + accessToken, HttpStatus.BAD_REQUEST, String.class);
+    }
+
+    @Test
+    @WithMockUser(value = "instuctor1", roles = "INSTRUCTOR")
+    public void testGetLecturePdfAttachmentsMerged_InvalidToken() throws Exception {
+        Long lectureId = createLectureWithLectureUnits(HttpStatus.CREATED);
+
+        // send request using the access token with invalid token
+        request.get("/api/files/attachments/lecture/" + lectureId + "/merge-pdf?access_token=random_non_valid_token", HttpStatus.FORBIDDEN, String.class);
+    }
+
+    @Test
+    @WithMockUser(value = "instuctor1", roles = "INSTRUCTOR")
+    public void testGetLecturePdfAttachmentsMerged() throws Exception {
+        Long lectureId = createLectureWithLectureUnits(HttpStatus.CREATED);
+
+        // get access token and then send request using the access token
+        String accessToken = request.get("/api/files/attachments/access-token/merge-pdf", HttpStatus.OK, String.class);
+        byte[] receivedFile = request.get("/api/files/attachments/lecture/" + lectureId + "/merge-pdf" + "?access_token=" + accessToken, HttpStatus.OK, byte[].class);
+
+        assertThat(receivedFile).isNotEmpty();
+        PDDocument mergedDoc = PDDocument.load(receivedFile);
+        assertEquals(5, mergedDoc.getNumberOfPages());
+    }
+
+    public Long createLectureWithLectureUnits(HttpStatus expectedStatus) throws Exception {
```

Comment:
Similar functions like this one and the one below exist in the `DatabaseUtilService` (e.g. `addLectureUnitsToLecture`). Can their functionality be used here to reduce code duplication? Can (some of) the code here can be moved to the other service so that other tests can use it in the future from a central location?